ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -700,23 +690,35 @@ void writeIndexSegmentToFile(Offset safeEndPoint) throws IOException, StoreExcep
             "IndexSegment : " + indexFile.getAbsolutePath() + " IO error while persisting index to disk", e,
             StoreErrorCodes.IOError);
       } finally {
-        writer.close();
         rwLock.readLock().unlock();
       }
       logger.trace("IndexSegment : {} completed writing index to file", indexFile.getAbsolutePath());
     }
   }
 
   /**
-   * Memory maps the segment of index. Optionally, it also persist the bloom filter to disk
-   * @param persistBloom True, if the bloom filter needs to be persisted. False otherwise.
-   * @throws IOException
-   * @throws StoreException
+   * Marks the segment as sealed. Optionally, it also persist the bloom filter to disk
+   * @throws IOException if there is any problem with I/O
+   * @throws StoreException if there are problems with the index
    */
-  void map(boolean persistBloom) throws IOException, StoreException {
-    RandomAccessFile raf = new RandomAccessFile(indexFile, "r");
+  void seal() throws IOException, StoreException {
+    if (sealed.compareAndSet(false, true)) {
```

Comment:
why is the `compareAndSet()` necessary? Why would this method be called more than once?

Reply:
No - removed the check